### PR TITLE
Ensure Blazor prompts only show once per open workspace.

### PR DESCRIPTION
**Pull request by JoeRobich**
@captainsafia This limits blazor prompts to only showing when a Blazor project is discovered and only once per workspace session.

addresses https://github.com/OmniSharp/omnisharp-vscode/pull/4470#issuecomment-812234878

### CODE REVIEW

File: src/omnisharp/server.ts [hunk 1]
```diff
@@ -106,6 +107,10 @@ export class OmniSharpServer {
         this.firstUpdateProject = true;
     }
 
+    public get sessionProperties() {
+        return this._sessionProperties;
```
Comment (JoeRobich) [inline src/omnisharp/server.ts:111]: Allow arbitrary properties to be tied to an OmniSharp session.

File: src/omnisharp/server.ts [hunk 2]
```diff
@@ -434,6 +439,9 @@ export class OmniSharpServer {
 
         let cleanupPromise: Promise<void>;
 
+        // Clear the session properties when the session ends.
+        this._sessionProperties = {};
```
Comment (JoeRobich) [inline src/omnisharp/server.ts:443]: Ensure these properties are cleared between sessions. (i.e. Reload OmniSharp is invoked)

File: src/omnisharp/utils.ts [hunk 3]
```diff
@@ -78,7 +78,6 @@ export async function requestProjectInformation(server: OmniSharpServer, request
 export async function requestWorkspaceInformation(server: OmniSharpServer) {
     const response = await server.makeRequest<protocol.WorkspaceInformationResponse>(protocol.Requests.Projects);
     if (response.MsBuild && response.MsBuild.Projects) {
-        const blazorDetectionEnabled = hasBlazorWebAssemblyDebugPrerequisites();
```
Comment (JoeRobich) [inline src/omnisharp/utils.ts:81]: Move the prerequisite check to after we know we have a Blazor project.

**Comment by NTaylorMullen** [comment]
Fixes https://github.com/dotnet/aspnetcore/issues/30834

**Review by captainsafia — APPROVED** [review]

**Pull request merged**